### PR TITLE
New server stuffs

### PR DESCRIPTION
**Pull request by GeoffreyWesthoff**
- Replaced Endpoints with Documentation
- Added ratelimit tracking, will send a webhook every 5 429's
- Auxim you shouldn't use inline styles so much SHUT UP MOM

### CODE REVIEW

File: utils/ratelimits.py [hunk 1]
```diff
@@ -65,6 +69,20 @@ def wrapper(*args, **kwargs):
...
+                try:
+                    ratelimit_reached = key['ratelimit_reached']
+                except KeyError:
+                    ratelimit_reached = 0
+                ratelimit_reached += 1
+                print(ratelimit_reached)
+                r.table('keys').get(auth).update({"ratelimit_reached": ratelimit_reached}).run(get_db())
+                if ratelimit_reached % 5 == 0 and 'webhook_url' in config:
+                    requests.post(config['webhook_url'], headers={'content-type': 'application/json'},
+                                  data=json.dumps({"embeds": [{
```
Comment (AndyTempel) [inline utils/ratelimits.py]: Just a little nit:
requests lib has an inbuilt json parameter that handles things for you as they should be.
```python
requests.post(config['webhook_url'], json={...})
```

File: utils/ratelimits.py [hunk 2]
```diff
@@ -65,6 +69,20 @@ def wrapper(*args, **kwargs):
                                         'X-RateLimit-Remaining': max_usage - usage - 1,
                                         'X-RateLimit-Reset': cache.expires_on(key['id'])}))
             else:
+                try:
+                    ratelimit_reached = key['ratelimit_reached']
```
Comment (devoxin) [inline utils/ratelimits.py]: ratelimit_reached = key.get('ratelimit_reached', 0) + 1

File: utils/ratelimits.py [hunk 3]
```diff
@@ -65,6 +69,20 @@ def wrapper(*args, **kwargs):
...
+                except KeyError:
+                    ratelimit_reached = 0
+                ratelimit_reached += 1
+                print(ratelimit_reached)
```
Comment (devoxin) [inline utils/ratelimits.py]: debug line?

File: utils/ratelimits.py [hunk 4]
```diff
@@ -65,6 +69,20 @@ def wrapper(*args, **kwargs):
...
+                r.table('keys').get(auth).update({"ratelimit_reached": ratelimit_reached}).run(get_db())
+                if ratelimit_reached % 5 == 0 and 'webhook_url' in config:
+                    requests.post(config['webhook_url'], headers={'content-type': 'application/json'},
+                                  data=json.dumps({"embeds": [{
+                                      "title": f"Application '{key['name']}' ratelimited 5 times!",
```
Comment (devoxin) [inline utils/ratelimits.py]: Is it worth showing the total amount of times they've hit ratelimit rather than 5 each time?

Comment (GeoffreyWesthoff) [reply utils/ratelimits.py]: Is does that in the embed, but the embed fires every 5 times a ratelimit is reached, so if someone is spamming you can :b::a:n

File: utils/ratelimits.py [hunk 5]
```diff
@@ -69,20 +69,14 @@ def wrapper(*args, **kwargs):
                                         'X-RateLimit-Remaining': max_usage - usage - 1,
                                         'X-RateLimit-Reset': cache.expires_on(key['id'])}))
             else:
-                try:
-                    ratelimit_reached = key['ratelimit_reached']
-                except KeyError:
-                    ratelimit_reached = 0
-                ratelimit_reached += 1
-                print(ratelimit_reached)
+                ratelimit_reached = key['ratelimit_reached', 0] + 1
```
Comment (devoxin) [inline utils/ratelimits.py]: `.get` 🌲

File: utils/ratelimits.py [hunk 6]
```diff
@@ -69,20 +69,14 @@ def wrapper(*args, **kwargs):
...
                 r.table('keys').get(auth).update({"ratelimit_reached": ratelimit_reached}).run(get_db())
                 if ratelimit_reached % 5 == 0 and 'webhook_url' in config:
-                    requests.post(config['webhook_url'], headers={'content-type': 'application/json'},
-                                  data=json.dumps({"embeds": [{
-                                      "title": f"Application '{key['name']}' ratelimited 5 times!",
-                                      "description": f"Owner: {key['owner']}\n"
-                                      f"Total: {ratelimit_reached}"}]}
-                                  ))
+                    requests.post(config['webhook_url'], json={"embeds": [{
+                                                              "title": f"Application '{key['name']}' ratelimited 5 times!",
```
Comment (devoxin) [inline utils/ratelimits.py:76]: dis lint? I feel like flake8 would complain that quotes should be one space over

File: utils/ratelimits.py [hunk 7]
```diff
@@ -69,20 +69,14 @@ def wrapper(*args, **kwargs):
...
+                                                              "description": f"Owner: {key['owner']}\n"
+                                                              f"Total: {ratelimit_reached}"}]}
+                                  )
```
Comment (devoxin) [inline utils/ratelimits.py]: lonely boi. Might as well whack it on the end of the above line

**Review by devoxin — CHANGES_REQUESTED** [review]

File: utils/ratelimits.py (hunk 4 above)
Comment (devoxin) [reply utils/ratelimits.py]: o ok

**Review by devoxin — CHANGES_REQUESTED** [review]
lint

**Review by devoxin — APPROVED** [review]

**Pull request merged**